trajopt_sco/src/osqp_interface.cpp

```diff
@@ -124,31 +124,34 @@ void OSQPModel::updateConstraints()
 {
   const size_t n = vars_.size();
   const size_t m = cnts_.size();
-  osqp_data_.m = m + n;
+  const int n_int = static_cast<int>(n);
+  const int m_int = static_cast<int>(m);
+
+  osqp_data_.m = static_cast<c_int>(m + n);
 
   Eigen::SparseMatrix<double> sm;
   Eigen::VectorXd v;
-  exprToEigen(cnt_exprs_, sm, v, n);
-  Eigen::SparseMatrix<double> sm_e(m + n, n);
+  exprToEigen(cnt_exprs_, sm, v, static_cast<int>(n));
+  Eigen::SparseMatrix<double> sm_e(n_int + m_int, n_int);
   Eigen::SparseMatrix<double> sm_e2 = sm;
-  sm.conservativeResize(m + n, Eigen::NoChange_t(n));
+  sm.conservativeResize(m_int + n_int, Eigen::NoChange_t(n));
 
   l_.clear();
   l_.resize(m + n, -OSQP_INFINITY);
   u_.clear();
   u_.resize(m + n, OSQP_INFINITY);
 
-  for (int i_cnt = 0; i_cnt < m; ++i_cnt)
+  for (int i_cnt = 0; i_cnt < m_int; ++i_cnt)
```

Comment:
could you use size_t as the type to avoid cast inside the for loop?

Reply:
I changed it. There are still static casts, but now there are 2 to int instead of 3 to size_t

Reply:
Oh, did not change much.